### PR TITLE
fwup: 1.5.2 -> 1.8.3

### LINKED ISSUE #110991: fwup is out-of-date
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: nixpkgs.fwup

current version: 1.5.2
desired version: 1.8.3

###### Notify maintainers

maintainers: @georgewhewell 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.8.3

Change log: https://github.com/fhunleth/fwup/blob/main/CHANGELOG.md

Fixes #110991

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111006` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fwup</li>
  </ul>
</details>

**Review by georgewhewell — APPROVED** [review]
LGTM

File: pkgs/tools/misc/fwup/default.nix [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+, makeWrapper
+, mtools
+, pkg-config
+, stdenv
+, unzip
+, which
+, xdelta
+, zip
+, zlib
+}:
```
Comment (danieldk) [inline pkgs/tools/misc/fwup/default.nix:21]: Inputs are typically sorted first by group: general use in derivation (`stdenv`, `lib`, fetchers, etc.), `nativeBuildInputs`, `buildInputs`, and then `checkInputs`. Then the inputs are sorted alphabetically per group.

I don't think it is a blocker, but IMO it's preferable to keep that order.

File: pkgs/tools/misc/fwup/default.nix [hunk 2]
```diff
@@ -1,26 +1,49 @@
...
     repo = "fwup";
     rev = "v${version}";
-    sha256 = "05sjdlh450hk474a44yr6kz9dzx72jfxpi1krxbd0pdizlmfypsg";
+    sha256 = "sha256-ayfcnIZ7MuBsCy1giwmY2D2C6AukwS+fevmXqGa4c1w=";
   };
 
   doCheck = true;
+
+  checkInputs = [ which xdelta ];
+
```
Comment (danieldk) [inline pkgs/tools/misc/fwup/default.nix:37]: It's more canonical to put `checkInputs` after build inputs (so here after `propagatedBuildInputs`). Yes, `doCheck` is a bit weirdly placed here, and should go after all inputs ;).

File: pkgs/tools/misc/fwup/default.nix [hunk 3]
```diff
@@ -1,26 +1,49 @@
...
   patches = lib.optional stdenv.isDarwin [ ./fix-testrunner-darwin.patch ];
 
   nativeBuildInputs = [ pkg-config autoreconfHook makeWrapper ];
+
   buildInputs = [ zlib lzma bzip2 libconfuse libsodium libarchive ]
     ++ lib.optionals stdenv.isDarwin [
       darwin.apple_sdk.frameworks.DiskArbitration
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fwup/default.nix:44]:
```suggestion
      DiskArbitration
```

darwin should not be used as an input. Please inherit DiskArbitration from top level.

File: pkgs/tools/misc/fwup/default.nix [hunk 4]
```diff
@@ -1,26 +1,49 @@
...
     rev = "v${version}";
-    sha256 = "05sjdlh450hk474a44yr6kz9dzx72jfxpi1krxbd0pdizlmfypsg";
+    sha256 = "sha256-ayfcnIZ7MuBsCy1giwmY2D2C6AukwS+fevmXqGa4c1w=";
   };
 
   doCheck = true;
+
+  checkInputs = [ which xdelta ];
+
   patches = lib.optional stdenv.isDarwin [ ./fix-testrunner-darwin.patch ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fwup/default.nix:38]: patches should no be applied conditional.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/fwup/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/tools/misc/fwup/default.nix]: Guess we are looking at `isort`-style sorting here. I can arrange the inputs in that way.

File: pkgs/tools/misc/fwup/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/tools/misc/fwup/default.nix]: Done

File: pkgs/tools/misc/fwup/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/tools/misc/fwup/default.nix]: Removed as it wouldn't affect other platform anyways.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111006` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fwup</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
fwup:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/misc/fwup/default.nix:34:15:

```
   |
34 |   patches = [ ./fix-testrunner-darwin.patch ];
   |               ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Unused argument: `darwin`.
Near pkgs/tools/misc/fwup/default.nix:5:3:

```
  |
5 | , darwin
  |   ^
```
</details>
</details>

File: pkgs/tools/misc/fwup/default.nix [hunk 5]
```diff
@@ -1,27 +1,69 @@
-{ stdenv, lib, fetchFromGitHub, autoreconfHook, makeWrapper, pkg-config
-, zlib, lzma, bzip2, mtools, dosfstools, zip, unzip, libconfuse, libsodium
-, libarchive, darwin, coreutils }:
+{ stdenv
+, lib
+, fetchFromGitHub
+, autoreconfHook
+, darwin
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fwup/default.nix:5]:
```suggestion
```

Comment (fabaff) [reply pkgs/tools/misc/fwup/default.nix]: Removed, thanks.

**Pull request merged**